RELEASE_PROCEDURE.md

```diff
@@ -0,0 +1,18 @@
+# Release procedure
+
+When you release fixed version of SonarQube SpotBugs Plugin, please follow these procedures.
+
+* create topic branch from `master` branch
+* change version number in `pom.xml` to stable version (e.g. `1.2.3`), then commit changes
+* change version number in `pom.xml` to next development SNAPSHOT version (e.g. `1.2.4-SNAPSHOT`), then commit changes
+* push your topic branch and propose a pull request
+* after merging your pull request, tag the commit which has stable version in `pom.xml`, and push this tag
+
+## Release to Maven Central
+
+When we push tag, the build result on Travis CI will be deployed to [SonaType Nexus](https://oss.sonatype.org/).
+Check [Sonatype official page](http://central.sonatype.org/pages/apache-maven.html) for detail.
+
+## Release to SonarQube plugin update center
+
+TBU
```

Comment:
What is our to do? Any document does sonar provides?

Reply:
Yes,

Requirements
============

If your plugin meets the following requirements, then you can ask SonarSource (on the [SonarQube Google Group](https://groups.google.com/forum/#!forum/sonarqube)) to reference your plugin in the [SonarQube Marketplace](/display/SONAR/Marketplace):

1.  Your plugin is open-source
    1.  Source is freely accessible
    2.  The license is a known FLOSS one (check [the list provided by the Open Source Initiative](http://opensource.org/licenses))
    3.  There's a public issue tracking system
    4.  Documentation is available online in English
    5.  Binaries of each version are accessible somewhere
2.  Releases follow open source conventions
    1.  A Release Candidate and feedback period are announced prior to each release.
    2.  A "feedback period closed" notice is sent just before the release.
    3.  For each release the following must be available:
        1.  release notes that reflect all significant changes in the version
        2.  plugin jar
3.  The key of your plugin should be:
    1.  short and unique
    2.  lowercase (no camelcase)
    3.  composed only of \[a-z0-9\]
    4.  examples: motionchart, php, scmactivity
4.  The description of your plugin must not be misleading in terms of content (the code needs to do pretty much what the name and description say it does). On initial entry into the Marketplace, SonarSource staff will test the plugin to verify reasonable functionality and quality. SonarSource staff must be provided with the necessary means to do this testing _without_ the need to sign any agreements or fill out any forms. Ideally, the means to do this testing will be made available to the community at large, with the same lack of restrictions.
5.  Your plugin does not compete with existing SonarSource products _(sorry, but we gotta pay the bills somehow)_
6.  It is analysed on [SonarCloud](https://sonarcloud.io) and the quality gate is green when doing a release
7.  It is compatible with the [platform requirements](/display/SONAR/Requirements) (e.g. it runs on the minimum listed JRE)
8.  Last but not least: your plugin must be aligned with the goal of the SonarQube platform: management of the technical debt and the quality of the code
    1.  To be more precise: every feature of SonarQube is tied to the code, so if your plugin provides data that can't be attached to a source or a test file, then there are chances that your plugin won't be accepted in the Marketplace

If your plugin meets these requirements, feel free to start a new thread on the [SonarQube Google Group](https://groups.google.com/forum/#!forum/sonarqube) requesting inclusion. This thread should include plugin:

*   description
*   home page url

as well as the elements of a "new release" email listed below.

We reserve the right to exclude from the Marketplace plugins that we feel would be a dis-service to the community.

Announcing new releases
=======================

When you've got a new release that should be published in the Marketplace, please start a new [SonarQube Google Group](https://groups.google.com/forum/#!forum/sonarqube) thread with the following information:

*   Subject: \[NEW RELEASE\] Plugin Name & version
*   Body contains:
    *   Short description: a few words about what's new in this version.
    *   Sonarqube compatibility: unchanged or specific versions.
    *   Download URL for the plugin binary
    *   Link to release notes
    *   Link to [SonarCloud](http://sonarcloud.io) project dashboard so that we can check the quality gate status
*   If it is the first release of the plugin, please mention that the plugin should be added to the [Plugin Library page](/display/PLUG/Other+Plugins)

Once this email is sent, someone from SonarSource will perform the manual steps to make the version available in the Marketplace.

Suggestions to manage your plugin development
=============================================

For instance, a project hosted on a GitHub repository can easily meet the requirements:

*   Sources are on Git - and you can easily configure them to be built by Travis CI
*   GitHub Issues can be used as a bug tracking system
*   GitHub Wiki can be used to write the documentation
*   GitHub Releases can be used to publish your binaries

You can obviously use the [SonarQube Google Group](https://groups.google.com/forum/#!forum/sonarqube) to ask for feedback on your plugin.

Source : https://docs.sonarqube.org/display/DEV/Deploying+to+the+Marketplace

Reply:
Thank you!